common/interfaces/user_interface/ui_events.h

```diff
@@ -0,0 +1,90 @@
+/**
+ * @file    ui_event.h
+ * @author  Cypherock X1 Team
+ * @brief   Title of the file.
+ *          Short description of the file
+ * @copyright Copyright (c) 2023 HODL TECH PTE LTD
+ * <br/> You may obtain a copy of license at <a href="https://mitcc.org/"
+ * target=_blank>https://mitcc.org/</a>
+ */
+#ifndef UI_EVENTS
+#define UI_EVENTS
+
+/*****************************************************************************
+ * INCLUDES
+ *****************************************************************************/
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+#include "assert_conf.h"
+#include "board.h"
+
+/*****************************************************************************
+ * MACROS AND DEFINES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * TYPEDEFS
+ *****************************************************************************/
+typedef enum {
+  UI_EVENT_CONFIRM = 1,
+  UI_EVENT_REJECT,
+  UI_EVENT_TEXT_INPUT,
+  UI_EVENT_LIST_CHOICE,
+  UI_EVENT_SKIP_EVENT
+} ui_event_types_t;
+
+typedef struct {
+  bool event_occured;
+  ui_event_types_t event_type;
+  char *text_ptr;
+  uint16_t list_selection;
+} ui_event_t;
+
+/*****************************************************************************
+ * EXPORTED VARIABLES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * GLOBAL FUNCTION PROTOTYPES
+ *****************************************************************************/
+
+/**
+ * @brief   Used to get the latest ui event occurance
+ * @arg     *ui_event_os_obj   `ui_event_t` object used to get the ui event
+ * details
+ */
+bool ui_get_and_reset_event(ui_event_t *ui_event_os_obj);
+
+/**
+ * @brief   Used to pass UI cofirm event to os event getter
+ *
+ * @return  returns true if event was set correctly and ui status was updated
+ */
+void ui_set_confirm_event();
```

Comment:
all the `ui_set_*` are only needed by the ui components. we can move the declaration to `ui_events_priv`.

and only keep `ui_get_and_reset_event` here.

Reply:
updated